src/thunder/renderer-backend.cpp

```diff
@@ -140,11 +140,25 @@ struct wpe_renderer_backend_egl_interface thunder_renderer_backend_egl_interface
     // destroy
     [](void* data)
     {
+        if (data != nullptr) {
+            Thunder::Compositor::IDisplay* display = static_cast<Thunder::Compositor::IDisplay*>(data);
```

Comment:
This is against the rules of reference counting. You get the interface offered, so the on offering you this interface has to have a Reference count. You do not maintain the interface and thus correctly do not do an AddRef (or QueryInterface) on this interface and thus you should not Release it as now you are releasing on beha;f of someone else...